scripts/run-tests.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+set -e
+
+root_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )/../" && pwd )
+if [ "${root_dir}" != "${PWD}" ]; then
+    echo "You must execute $(basename $0) from the root directory: ${root_dir}"
+    exit 1
+fi
+
+WORKSPACE=${PWD}/tests
+MENDER_CONVERT_DIR=${PWD}
+
+BBB_DEBIAN_IMAGE="bone-debian-9.5-iot-armhf-2018-10-07-4gb.img"
+BBB_DEBIAN_IMAGE_URL="http://debian.beagleboard.org/images/${BBB_DEBIAN_IMAGE}.xz"
+
+RASPBIAN_IMAGE="2019-04-08-raspbian-stretch-lite"
+RASPBIAN_IMAGE_URL="https://downloads.raspberrypi.org/raspbian_lite/images/raspbian_lite-2019-04-09/${RASPBIAN_IMAGE}.zip"
+
+TINKER_IMAGE="20170417-tinker-board-linaro-stretch-alip-v1.8"
+TINKER_IMAGE_URL="http://dlcdnet.asus.com/pub/ASUS/mb/Linux/Tinker_Board_2GB/${TINKER_IMAGE}.zip"
+
+UBUNTU_IMAGE="Ubuntu-Bionic-x86-64.img"
+UBUNTU_IMAGE_URL="https://d1b0l86ne08fsf.cloudfront.net/mender-convert/images/${UBUNTU_IMAGE}.gz"
+
+MENDER_ACCEPTANCE_URL="https://raw.githubusercontent.com/mendersoftware/meta-mender/master/tests/acceptance"
+
+# Some distros do not have /sbin in path for "normal users"
+export PATH="${PATH}:/sbin"
+
+convert_and_test() {
+    device_type=$1
+    artifact_name=$2
+    image_url=$3
+    image_name=$4
+    image_name_compressed=$5
+    config=$6 # Optional
+
+    wget -N ${image_url} -P input/
+
+    echo "Extracting: ${image_name_compressed}"
+    case "${image_name_compressed}" in
+    *.gz)
+        gunzip -f input/${image_name_compressed}
+        ;;
+    *.zip)
+        cd input
+        unzip -o ${image_name_compressed}
+        cd -
+      ;;
+    *.xz)
+        xz -d -f input/${image_name_compressed}
+        ;;
+    *)
+        echo "Unknown image type: ${image_name_compressed}"
+        exit 1
+    esac
+
+    rm -f /tmp/test_config
+
+    if [ -n "${config}" ]; then
+        cp ${config} /tmp/test_config
+    fi
+
+    # Disable compression of disk image when testing, otherwise we need to
+    # unpack each image we test which is time consuming
+    echo "MENDER_COMPRESS_DISK_IMAGE=n" >> /tmp/test_config
+
+    echo "Configuration used:"
+    cat /tmp/test_config
+
+    MENDER_ARTIFACT_NAME=${artifact_name} ./mender-convert \
+        --disk-image input/${image_name} \
+        --config /tmp/test_config
```

Comment:
I will update the test script to run `docker-mender-convert`, and I agree that the `docker-mender-convert` should be the official way of running the tool.

But we really should still support a way of running this tool without docker even though this can be untested (officially), as I know of a case where a Docker container will not work, e.g managing LVM (at least I could not get it work last time I worked with it), where the conversion process had to be done native.

Reply:
I agree in keeping the non-docker support as well.